tests/sys/psa_crypto/main.c

```diff
@@ -111,6 +111,41 @@ static void test_hash_interleaved(void)
     TEST_ASSERT_PSA_CONTINUE(psa_hash_abort(&op2));
 }
 
+/**
+ * Creating keys with the usage flags PSA_KEY_USAGE_SIGN_HASH/PSA_KEY_USAGE_VERIFY_HASH
+ * should automatically set the usage flags PSA_KEY_USAGE_SIGN_MESSAGE/PSA_KEY_USAGE_VERIFY_MESSAGE
+ * on the key.
+ */
+static void test_key_usage_flags(void)
+{
+    psa_key_attributes_t attributes = psa_key_attributes_init();
+    psa_key_attributes_t key_attrs;
+    const uint8_t key[32] = { 0 };
+    psa_key_usage_t key_usage;
+    psa_key_id_t key_id;
+
+    psa_set_key_algorithm(&attributes, PSA_ALG_HMAC(PSA_ALG_SHA_512));
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH);
+    psa_set_key_bits(&attributes, PSA_BYTES_TO_BITS(sizeof(key)));
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_HMAC);
+
+    TEST_ASSERT_PSA_RETURN(psa_crypto_init());
+    TEST_ASSERT_PSA_RETURN(psa_import_key(&attributes, key, sizeof(key), &key_id));
```

Comment:
This only tests the behavior when importing a key, not when generating one. Maybe you want to add that too?

Reply:
Good point.

Reply:
Thanks for adding that, too!